src/logger.rs

```diff
@@ -58,9 +58,17 @@ impl Logger {
 }
 
 impl log::Log for Logger {
-    fn enabled(&self, _: &Metadata) -> bool {
-        // We set to true such that each logs arrives in the log function and then we filter
-        true
+    fn enabled(&self, metadata: &Metadata) -> bool {
+        // We filter depending on the current log level.
+        metadata.level()
+            <= match config::LOG_LEVEL {
+                Some("trace") => LevelFilter::Trace,
+                Some("debug") => LevelFilter::Debug,
+                Some("info") => LevelFilter::Info,
+                Some("warn") => LevelFilter::Warn,
+                Some("error") => LevelFilter::Error,
+                Some(&_) | None => LevelFilter::Off,
+            }
```

Comment:
Unfortunately it's not that simple, because we can enable per-module level. You have to use the `filter_by_global_level` (which does what you have here) and `filter_by_module` functions and do an or between the two (see `log` implementation bellow).

Reply:
I see, a problem here is that the `enabled` take a `&Metadata` object and not a `&Record`. And so I have no idea how to get the target.

Reply:
Hmmm, actually it looks like those two `filter_by_*` functions only use the record from the metadata, so we could change their signatures to take a record (and pass it from `metadata.record` in `log`).

Reply:
Actually, I think it is the opposite: metadata come from the record, and trait `log::Log` define `enabled` function using only metadata.